### ISSUE TITLE
ImportError: cannot import name 'update_data_root' from 'mmdet.utils'

### ISSUE THREAD

**Issue by firaterdem** [open-mmlab/mmdetection#7581]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**
I got an import error message when I ran the train code.

**Reproduction**

1. What command or script did you run?

```none
C:\Users\firat\Desktop\mmdetection-master>python tools\train.py configs\balloon\config_file.py
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?

- I have modified the config file by changing directories.

3. What dataset did you use?

- I have created a custom dataset.

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.

'tail' is not recognized as an internal or external command,
operable program or batch file.
'gcc' is not recognized as an internal or external command,
operable program or batch file.
sys.platform: win32
Python: 3.7.11 (default, Jul 27 2021, 09:42:29) [MSC v.1916 64 bit (AMD64)]
CUDA available: True
GPU 0: NVIDIA GeForce GTX 1060
CUDA_HOME: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1
NVCC: Not Available
GCC: n/a
PyTorch: 1.6.0
PyTorch compiling details: PyTorch built with:
  - C++ Version: 199711
  - MSVC 192628806
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191125 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.5.0 (Git Hash e2ac1fac44c5078ca927cb9b90e1b3066a0b2ed0)
  - OpenMP 2019
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.4
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -openmp:experimental -DNDEBUG -DUSE_FBGEMM -DUSE_VULKAN_WRAPPER, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,

TorchVision: 0.7.0
OpenCV: 4.5.5
MMCV: 1.4.4
MMCV Compiler: MSVC 192930137
MMCV CUDA Compiler: 10.1
MMDetection: 2.22.0+unknown

2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    I have applied these instructions:
    https://mmdetection.readthedocs.io/en/v2.21.0/get_started.html#installation
    
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)
 C:\Users\firat\Desktop\mmdetection-master>PATH
PATH=C:\Users\firat\anaconda3\envs\open-mmlab;C:\Users\firat\anaconda3\envs\open-mmlab\Library\mingw-w64\bin;C:\Users\firat\anaconda3\envs\open-mmlab\Library\usr\bin;C:\Users\firat\anaconda3\envs\open-mmlab\Library\bin;C:\Users\firat\anaconda3\envs\open-mmlab\Scripts;C:\Users\firat\anaconda3\envs\open-mmlab\bin;C:\Users\firat\anaconda3\condabin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\bin;C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1\libnvvp;.;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\MATLAB\R2020a\runtime\win64;C:\Program Files\MATLAB\R2020a\bin;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit;C:\Program Files\CODA\bin;C:\Program Files\HARP\bin;C:\Program Files (x86)\Calibre2;C:\Program Files\SafeNet\Authentication\SAC\x64;C:\Program Files\SafeNet\Authentication\SAC\x32;C:\Program Files\NVIDIA Corporation\Nsight Compute 2019.4.0;C:\Program Files\Hexagon\ERDAS IMAGINE 2015;C:\Program Files\PuTTY;C:\Users\firat\AppData\Local\Programs\Python\Python37\Scripts;C:\Users\firat\AppData\Local\Programs\Python\Python37;C:\Users\firat\anaconda3;C:\Users\firat\anaconda3\Library\mingw-w64\bin;C:\Users\firat\anaconda3\Library\usr\bin;C:\Users\firat\anaconda3\Library\bin;C:\Users\firat\anaconda3\Scripts;C:\Users\firat\AppData\Local\Microsoft\WindowsApps;.;C:\Program Files\snap\bin

**Error traceback**
If applicable, paste the error trackback here.

```none
(open-mmlab) C:\Users\firat\Desktop\mmdetection-master>python tools\train.py configs\balloon\config_file.py
Traceback (most recent call last):
  File "tools\train.py", line 20, in <module>
    from mmdet.utils import (collect_env, get_root_logger, setup_multi_processes,
ImportError: cannot import name 'update_data_root' from 'mmdet.utils' (C:\Users\firat\anaconda3\envs\open-mmlab\lib\site-packages\mmdet\utils\__init__.py)
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by jbwang1997** [comment]
Hi @firaterdem. The `update_data_root` function is recently merged in mmdet v2.23.0. But according to the information above, you use the mmdet v2.22.0. You can try to use the latest mmdetection.

**Comment by firaterdem** [comment]
I have solved the problem by installing the version you mentioned with "pip install mmdet==2.23.0". Thank you for your help.
